Doberman/AlarmNode.py

```diff
@@ -0,0 +1,124 @@
+import Doberman
+
+
+class AlarmNode(Doberman.Node):
+    """
+    An empty base class to handle database access
+    """
+    def setup(self, **kwargs):
+        super().setup(**kwargs)
+        self.description = kwargs['description']
+        self.sensor = kwargs['sensor']
+        self._log_alarm = kwargs['log_alarm']
+        self.escalation_config = kwargs['escalation_config']
+        self.escalation_level = 0
+        self.base_level = kwargs['base_level']
+        self.auto_silence_duration = kwargs.get('silence_duration', 5*60)
+        self.alarm_start = None
+        self.hash = None
+
+    def escalate(self):
+        """
+        Do we escalate? This function decides this
+        """
+        if self.alarm_start is None:
+            # not sure how we got here, but there's nothing to escalate
+            return
+        time_since_start = time.time() - self.alarm_start
+        if time_since_start > self.escalation_config[max(self.base_level + self.escalation_level, len(self.escalation_config)-1)]:
```

Comment:
Is there a reason why you want to repeatedly "re-escalate" to the highest level?

Reply:
The `if` just checks to see if we've been in an alarmed state for long enough that escalation is necessary. The rest of the line makes sure that we don't try to look past the end of the `escalation_config` list. It could get split into two lines, that would probably make this clearer.

Reply:
On closer inspection, that `max` is supposed to be a `min`. Good catch